src/engine/Util/Clock.ts

```diff
@@ -94,13 +99,26 @@ export abstract class Clock {
    * @param cb callback to fire
    * @param timeoutMs Optionally specify a timeout in milliseconds from now, default is 0ms which means the next possible tick
    * @param timing Optionally specify a timeout in milliseconds from now, default is 0ms which means the next possible tick
+   * @returns A unique identifier that can be used to clear the scheduled callback with {@apilink clearSchedule}
    */
-  public schedule(cb: (elapsed: number) => any, timeoutMs: number = 0, timing: ScheduledCallbackTiming = 'preframe') {
+  public schedule(cb: (elapsed: number) => any, timeoutMs: number = 0, timing: ScheduledCallbackTiming = 'preframe'): ScheduleId {
     // Scheduled based on internal elapsed time
     const scheduledTime = this._totalElapsed + timeoutMs;
-    this._scheduledCbs.push([cb, scheduledTime, timing]);
+    const id = this._nextScheduleId++;
+    this._scheduledCbs.push([id, cb, scheduledTime, timing]);
+    return id;
   }
 
+  /**
+   * Clears a scheduled callback using the ID returned from {@apilink schedule}
+   * @param id The ID of the scheduled callback to clear
+   */
+  public clearSchedule(id: ScheduleId): void {
+    const index = this._scheduledCbs.findIndex(([scheduleId]) => scheduleId === id);
```

Comment:
only concern: tight loop performance? if user clears every frame or something...

to fix would need new data structure?

Reply:
Truth, if there were a LOT of scheduled callbacks this could start to be bad. I vote we keep this as is until it's a problem, my intuition says we'll be in the 10's of callbacks which will be low overhead to scan